src/main/scala/firrtl/passes/Checks.scala

```diff
@@ -83,24 +83,31 @@ trait CheckHighFormLike {
           errors.append(new IncorrectNumConstsException(info, mname, e.op.toString, nc))
       }
 
+      def nonNegativeConsts(): Unit = {
+        e.consts.map(_.toInt).filter(_ < 0).foreach {
```

Comment:
While this is huge improvement that I am happy to approve as is, it does leave a (less-common) bug open:
```
circuit m:
  module m:
    input i1: UInt<1>
    input i2: UInt<1>
    output o: UInt<2>
    o <= bits(add(i1, i2), 0, -11823845700)
```
You get
```
[error] Exception in thread "main" firrtl.passes.CheckHighFormLike$LsbLargerThanMsbException:  @[:test.fir@6.4]: [module m] Primop bits lsb 1061056188 > 0.
```

Unfortunately, `BigInt.toInt` just truncates into a 32-bit Int 👿

Reply:
True. I just fixed that and the other pre-existing Int-wrapping bug (for LSB > MSB check).